core/config/toml/types_test.go

```diff
@@ -180,3 +181,126 @@ func TestDatabaseSecrets_ValidateConfig(t *testing.T) {
 		})
 	}
 }
+func TestTracing_ValidateCollectorTarget(t *testing.T) {
+	tests := []struct {
+		name            string
+		collectorTarget *string
+		wantErr         bool
+		errMsg          string
+	}{
+		{
+			name:            "valid localhost address",
+			collectorTarget: stringPtr("localhost:4317"),
+			wantErr:         false,
+		},
+		{
+			name:            "valid docker address",
+			collectorTarget: stringPtr("otel-collector:4317"),
+			wantErr:         false,
+		},
+		{
+			name:            "valid IP address",
+			collectorTarget: stringPtr("192.168.1.1:4317"),
+			wantErr:         false,
+		},
+		{
+			name:            "invalid address",
+			collectorTarget: stringPtr("invalid address"),
+			wantErr:         true,
+			errMsg:          "CollectorTarget: invalid value (invalid address): must be a valid URI",
+		},
+		{
+			name:            "nil CollectorTarget",
+			collectorTarget: stringPtr(""),
+			wantErr:         true,
+			errMsg:          "CollectorTarget: invalid value (): must be a valid URI",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tracing := &Tracing{
+				CollectorTarget: tt.collectorTarget,
+			}
+
+			err := tracing.ValidateConfig()
+
+			if tt.wantErr {
+				assert.Error(t, err)
+				assert.Equal(t, tt.errMsg, err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+// stringPtr is a utility function for converting a string to a string pointer.
+func stringPtr(s string) *string {
+	return &s
+}
```

Comment:
You can consolidate these helpers in to one generic version:
```suggestion
func ptr[T any](t T) *T { return &t }
```